src/NBitcoin/FeeRate.cs

```diff
@@ -41,7 +41,7 @@ public FeeRate(Money feePaid, int size)
             if(feePaid.Satoshi < 0)
                 throw new ArgumentOutOfRangeException("feePaid");
             if(size > 0)
-                this._FeePerK = feePaid / size * 1000;
+                this._FeePerK = (long)(feePaid.Satoshi / (decimal)size * 1000);
```

Comment:
```
this._FeePerK = (long)(feePaid.Satoshi / ((decimal)size * 1000));
```

Brackets missing. Otherwise it doesn't fix the problem. 

Ps
Please add a test for it.

Reply:
yes a test is a good idea

Reply:
why you say it doesn't fix?

Reply:
agree on test

Reply:
well after testing it, it's working as intended.
Actually it will brake if you max out (long.MaxValue) left value because if size < 1000 it will overflow no matter where you put brackets

Reply:
Ah, yeah, right. 
Stupid mistake from me.